src/views/Admin.vue

```diff
@@ -341,11 +345,92 @@
 </template>
 
 <script lang="ts">
-import { defineComponent } from "vue";
+import { defineComponent, ref, onMounted } from "vue";
+import { TeamRecord, getTeam } from "@/api/teams";
+import { PaginatedList } from "@/api/base";
+import { listMembers, approveMember, MemberRecord } from "@/api/users";
+import store from "@/store";
 
 export default defineComponent({
-  name: "Admin"
+  name: "Admin",
   // components: {
   // }
+  props: {
+    teamId: {
+      type: String,
+      required: true
+    }
+  },
+  setup(props) {
+    const teamRecord = ref({} as TeamRecord);
+    const membersList = ref({} as PaginatedList<MemberRecord>);
+    const listPage = ref(1);
+    const teamId = parseInt(props.teamId);
+    const unapprovedList = ref({} as Array<MemberRecord>);
+
+    const getTeamMembers = async () => {
+      const axiosInstance = store.getters.generateAxiosInstance;
+      const team = teamRecord.value;
+      if (team) {
+        console.log("found team :)");
+        membersList.value = await listMembers(
+          axiosInstance,
+          teamId,
+          listPage.value
+        );
+
+        // create a array of unapproved users by filtering memberList
+        unapprovedList.value = membersList.value.results.filter(member => member.role === 'UA');
+        // console.log(unapprovedList.value);
+
+      } else {
+        console.log("no team :(");
+      }
+    };
+
+    const getTeamRecord = async () => {
+      const axiosInstance = store.getters.generateAxiosInstance;
+      const teamId = parseInt(props.teamId);
+      const team = await getTeam(axiosInstance, teamId);
+      if (team) {
```

Comment:
getTeam will either return a team record or throw an exception. Best to use try/catch instead of checking if team is defined.